File: data/EXP01/golang_no_system_var/2280/2280.output_1.go
func minimumLines(stockPrices [][]int) int {
    n := len(stockPrices)
    if n == 0 {
        return 0
    }
    
    lines := 1
    prevSlope := float64(stockPrices[1][1] - stockPrices[0][1]) / float64(stockPrices[1][0] - stockPrices[0][0])
    
    for i := 2; i < n; i++ {
        slope := float64(stockPrices[i][1] - stockPrices[i-1][1]) / float64(stockPrices[i][0] - stockPrices[i-1][0])
        
        if slope != prevSlope {
            lines++
            prevSlope = slope
        }
    }
    
    return lines
}

File: data/EXP01/python/2280/2280.output_1.py
# @lc app=leetcode id=2280 lang=python3
class Solution:
    def minimumLines(self, stockPrices):
        count = 1  # start with one line
        prev_price = stockPrices[0][1]  # initialize the previous price

        for i in range(1, len(stockPrices)):
            curr_price = stockPrices[i][1]  # get the current price

            if curr_price != prev_price:
                count += 1  # increment the count of lines
                prev_price = curr_price  # update the previous price

        return count